SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -28,34 +28,38 @@ void SpatialLoadBalanceEnforcer::OnAuthorityIntentComponentUpdated(const Worker_
 	}
 }
 
-// This is called whenever this worker becomes authoritative for the ACL component on an entity.
+// This is called whenever this worker becomes authoritative over the ACL component on an entity.
 // It is now this worker's responsibility to make sure that the ACL reflects the current intent,
 // which may have been received before this call.
 void SpatialLoadBalanceEnforcer::AuthorityChanged(const Worker_AuthorityChangeOp& AuthOp)
 {
-	if (AuthOp.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID &&
-		AuthOp.authority == WORKER_AUTHORITY_AUTHORITATIVE)
+	if (!(AuthOp.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID &&
+		AuthOp.authority == WORKER_AUTHORITY_AUTHORITATIVE))
 	{
-		const SpatialGDK::AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(AuthOp.entity_id);
-		if (AuthorityIntentComponent == nullptr)
-		{
-			// TODO(zoning): There are still some entities being created without an authority intent component.
-			// For example, the Unreal created worker entities don't have one. Even though those won't be able to
-			// transition, we should have the intent component on them for completeness.
-		 	UE_LOG(LogSpatialLoadBalanceEnforcer, Log, TEXT("Requested authority change for entity without AuthorityIntent component. EntityId: %lld"), AuthOp.entity_id);
-			return;
-		}
+		UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("Loadbalancer informed of authority change for entity %lld that was not a delegation of the ACL component."), AuthOp.entity_id);
```

Comment:
Changed this and added a warning if it gets triggered. The doc for the function indicated it would be called only when the ACL component was delegated, but it was actually called on every authority change, so changed that.